pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -373,6 +373,31 @@ static char *PyTimeToJSON(JSOBJ _obj, JSONTypeContext *tc, size_t *outLen) {
   return outValue;
 }
 
+static char *PyDecimalToUTF8Callback(JSOBJ _obj, JSONTypeContext *tc,
+                                     size_t *len) {
+  PyObject *obj = (PyObject *)_obj;
+  PyObject *str = PyObject_Str(obj);
+  if (str == NULL) {
+    *len = 0;
+    if (!PyErr_Occurred()) {
+      PyErr_SetString(PyExc_ValueError, "Failed to convert decimal");
+    }
+    ((JSONObjectEncoder *)tc->encoder)->errorMsg = "";
+    return NULL;
+  }
+  if (PyUnicode_Check(str)) {
+    PyObject *tmp = str;
+    str = PyUnicode_AsUTF8String(str);
```

Comment:
If our object is truly a decimal to begin with, we shouldn't need to take the unicode object and try to encode to utf-8; there shouldn't be any non-ascii characters contained within a decimal string.

Is there a `PyDecimal_Check` function or something exposed by the decimal module to get a string from that type of object? Generally seems like that could help simplify this